File: frontend-react/project/src/features/processing/utils/format.ts
// Small display helpers shared by the Processing pages.

const CURRENCY_PREFIX = 'Rs.'; // same convention as the Sales pages

const moneyFormat = new Intl.NumberFormat('en-US', { minimumFractionDigits: 2, maximumFractionDigits: 2 });
const weightFormat = new Intl.NumberFormat('en-US', { minimumFractionDigits: 0, maximumFractionDigits: 2 });

export const formatMoney = (amount: number): string => `${CURRENCY_PREFIX} ${moneyFormat.format(amount)}`;

export const formatKg = (kg: number): string => `${weightFormat.format(kg)} kg`;

/**
 * ASP.NET serialises UTC DateTimes without a zone when their Kind is Unspecified. Treat any
 * zone-less timestamp as UTC so it is not silently shown in the wrong time.
 */
const parseApiDate = (iso: string): Date => {
  const hasZone = /(Z|[+-]\d{2}:?\d{2})$/i.test(iso);
  return new Date(hasZone ? iso : `${iso}Z`);
};

export const formatDateTime = (iso: string | null | undefined): string => {
  if (!iso) return '—';
  const d = parseApiDate(iso);
  return Number.isNaN(d.getTime())
    ? '—'
    : d.toLocaleString(undefined, { year: 'numeric', month: 'short', day: 'numeric', hour: '2-digit', minute: '2-digit' });
};

export const formatDate = (iso: string | null | undefined): string => {
  if (!iso) return '—';
  const d = parseApiDate(iso);
  return Number.isNaN(d.getTime()) ? '—' : d.toLocaleDateString(undefined, { year: 'numeric', month: 'short', day: 'numeric' });
};

/** "3fa85f64…" — enough of a GUID to recognise it without filling the screen. */
export const shortId = (id: string | null | undefined): string => (id ? `${id.slice(0, 8)}…` : '—');

/** Signed difference with an explicit +, for weight discrepancies. */
export const formatSignedKg = (kg: number): string => `${kg > 0 ? '+' : ''}${weightFormat.format(kg)} kg`;

File: mobile_flutter/lib/core/utils/format.dart
import 'package:intl/intl.dart';

/// Display helpers — the same conventions as the web app's utils/format.ts.
abstract final class Format {
  static final _money = NumberFormat('#,##0.00', 'en_US');
  static final _weight = NumberFormat('#,##0.##', 'en_US');
  static final _dateTime = DateFormat('MMM d, y, h:mm a');
  static final _date = DateFormat('MMM d, y');

  /// "Rs. 1,234.50" — same currency prefix as the Sales and Processing pages.
  static String money(num amount) => 'Rs. ${_money.format(amount)}';

  static String kg(num kg) => '${_weight.format(kg)} kg';

  /// Signed difference with an explicit +, for weight discrepancies.
  static String signedKg(num kg) => '${kg > 0 ? '+' : ''}${_weight.format(kg)} kg';

  /// ASP.NET sends UTC timestamps without a zone when their Kind is Unspecified. Treat any
  /// zone-less timestamp as UTC so it is not shown in the wrong time.
  static DateTime? parseApiDate(String? iso) {
    if (iso == null || iso.isEmpty) return null;
    final hasZone = RegExp(r'(Z|[+-]\d{2}:?\d{2})$', caseSensitive: false).hasMatch(iso);
    return DateTime.tryParse(hasZone ? iso : '${iso}Z')?.toLocal();
  }

  static String dateTime(String? iso) {
    final d = parseApiDate(iso);
    return d == null ? '—' : _dateTime.format(d);
  }

  static String date(String? iso) {
    final d = parseApiDate(iso);
    return d == null ? '—' : _date.format(d);
  }

  /// "3fa85f64…" — enough of a GUID to recognise it without filling the screen.
  static String shortId(String? id) => id == null || id.isEmpty ? '—' : '${id.substring(0, id.length < 8 ? id.length : 8)}…';
}
